### PR TITLE
Fix eth_sendTransaction and add more RPC methods

### PR DESCRIPTION
**Pull request by dahu33**
This PR tries to make `headless-web3-provider` even more awesome:
1) fixed the `eth_sendTransaction` method which was broken when passing the `type` or `gas` parameters (JSON-RPC transaction need to be converted to an ethers.js transaction).
2) added tests for the above
3) added some missing pass-through RPC methods

@cawabunga if you could have a look that would be great! Thanks!

### CODE REVIEW

File: tests/e2e/metamask.spec.ts [hunk 1]
```diff
@@ -117,6 +118,44 @@ test('deploy a token', async ({ page }) => {
...
+}
+
+test('send legacy transaction', async ({ page, accounts }) => {
+  await page.locator('text=Connect').click()
+  await wallet.authorize(Web3RequestKind.RequestAccounts)
+
+  const nonceBefore = await getTransactionCount(page, accounts[0])
+
+  await page.locator('text=Send Legacy Transaction').click()
+  await wallet.authorize(Web3RequestKind.SendTransaction)
```
Comment (cawabunga) [inline tests/e2e/metamask.spec.ts:139]: Let's get `nonceBefore` right before authorizing the request, so we will be sure that transaction is not sent before it.

File: src/Web3ProviderBackend.ts [hunk 2]
```diff
@@ -388,3 +406,34 @@ function without<T>(list: T[], item: T): T[] {
...
+
+// Allowed keys for a JSON-RPC transaction as defined in:
+// https://ethereum.github.io/execution-apis/api-documentation/
+const allowedTransactionKeys =  ['accessList', 'chainId', 'data', 'from', 'gas', 'gasPrice', 'maxFeePerGas',
+  'maxPriorityFeePerGas', 'nonce', 'to', 'type', 'value']
+
+// Convert a JSON-RPC transaction to an ethers.js transaction.
+// The reverse of this function can be found in the ethers.js library:
+// https://github.com/ethers-io/ethers.js/blob/v5.7.2/packages/providers/src.ts/json-rpc-provider.ts#L701
+function convertTransaction(tx: { [key: string]: any }): ethers.providers.TransactionRequest {
```
Comment (cawabunga) [inline src/Web3ProviderBackend.ts:418]: Even though return type is specified, it is not very clear what the input and the output are. Maybe something like `convertTxObjectToEthersTxRequest`?

File: package.json [hunk 3]
```diff
@@ -35,7 +35,7 @@
   "devDependencies": {
     "@parcel/packager-ts": "2.7.0",
     "@parcel/transformer-typescript-types": "^2.7.0",
-    "@playwright/test": "^1.26.1",
+    "@playwright/test": "^1.38.0",
```
Comment (cawabunga) [inline package.json:38]: If there is no real necessity, let's skip upgrading.

File: playwright.config.ts [hunk 4]
```diff
@@ -21,14 +21,13 @@ const config: PlaywrightTestConfig = {
      */
     timeout: 5000,
   },
-  /* Run tests in files in parallel */
-  fullyParallel: true,
+  /* Disable parallelism to avoid the `nonce has already been used` issue */
```
Comment (cawabunga) [inline playwright.config.ts:24]: I fixed that in different way: https://github.com/cawabunga/headless-web3-provider/pull/13

**Review by cawabunga — CHANGES_REQUESTED** [review]

**Comment by dahu33** [comment]
@cawabunga thanks for the review, I've addressed your feedback, could you please have another look?

**Review by cawabunga — APPROVED** [review]
> @cawabunga thanks for the review, I've addressed your feedback, could you please have another look?

All good, thanks for the PR! 🙂

**Comment by cawabunga** [comment]
Published as [0.2.3](https://www.npmjs.com/package/headless-web3-provider/v/0.2.3)

**Pull request merged**